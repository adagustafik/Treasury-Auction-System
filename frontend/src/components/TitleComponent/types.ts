export interface IProps {
  pageTitle: string;
}
