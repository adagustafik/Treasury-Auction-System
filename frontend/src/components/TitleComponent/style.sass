@use '../../App'

.page-title-box
    text-align: center

.page-title-box h1
    margin: 15px auto 
