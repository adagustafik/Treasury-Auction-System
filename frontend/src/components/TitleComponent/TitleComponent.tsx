import React from "react";
import { IProps } from "./types";
import "./style.sass";

export function TitleComponent({ pageTitle }: IProps) {
  return (
    <div className="page-title-box">
      <h1>{pageTitle}</h1>
    </div>
  );
}
