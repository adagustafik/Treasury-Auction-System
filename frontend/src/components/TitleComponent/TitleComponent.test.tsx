import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { TitleComponent } from "./TitleComponent";

it("renders the TitleComponent correctly", () => {
  const registerForm = renderer
    .create(
      <BrowserRouter>
        <TitleComponent pageTitle="A page Title" />
      </BrowserRouter>
    )
    .toJSON();
  expect(registerForm).toMatchSnapshot();
});
