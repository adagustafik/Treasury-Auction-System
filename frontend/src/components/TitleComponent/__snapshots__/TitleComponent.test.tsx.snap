// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the TitleComponent correctly 1`] = `
<div
  className="page-title-box"
>
  <h1>
    A page Title
  </h1>
</div>
`;
